packages/teams-js/src/public/media.ts

```diff
@@ -691,38 +705,35 @@ export namespace media {
    * @param callback - callback to invoke after scanning the barcode
    * @param config - optional input configuration to customize the barcode scanning experience
    */
-  export function scanBarCode(callback: (error: SdkError, decodedText: string) => void, config?: BarCodeConfig): void {
-    if (!callback) {
-      throw new Error('[media.scanBarCode] Callback cannot be null');
-    }
-    ensureInitialized(FrameContexts.content, FrameContexts.task);
-
-    if (
-      GlobalVars.hostClientType === HostClientType.desktop ||
-      GlobalVars.hostClientType === HostClientType.web ||
-      GlobalVars.hostClientType === HostClientType.rigel ||
-      GlobalVars.hostClientType === HostClientType.teamsRoomsWindows ||
-      GlobalVars.hostClientType === HostClientType.teamsRoomsAndroid ||
-      GlobalVars.hostClientType === HostClientType.teamsPhones ||
-      GlobalVars.hostClientType === HostClientType.teamsDisplays
-    ) {
-      const notSupportedError: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
-      callback(notSupportedError, null);
-      return;
-    }
+  export function scanBarCode(callback: (error: SdkError, decodedText: string) => void, config?: BarCodeConfig): void;
```

Comment:
I think it would make more sense to just keep a single function signature here (the original one that returns void with a callback), mark it as deprecated, and then have it call the new promisified function in the barcode capability